src/Logary/Metric/MetricRegistry.fs

```diff
@@ -0,0 +1,24 @@
+namespace Logary.Metric
+
+open System.Collections.Concurrent
+
+/// used for register metric, and can export metric infos
+type MetricRegistry() =
+  let metricBackStore = new ConcurrentDictionary<string, MetricExporter>()
```

Comment:
Do we need it to be concurrent? The configuration API is not intended to be called concurrently?

Reply:
it depend on user's usage, mertic can be register throuth a `LogManager`'s instance, or use a static way to use the default registry. each metric with the same name is singleton semantically. consider this semantically, it implements with thread safe.
usage: 
```f#
module A =

  let itemsInQueueGauge = GaugeConf.create "Items_In_Queue" "help info with metric name" |> LogManager.DefaultMetricRegistry.registerMetricWithNoLables

  let enqueue (items: #seq<_>) =
    let itemCount = items |> Seq.length |> float
    itemsInQueueGauge.inc itemCount
    // enqueue items logic
    

type B(logManager:LogManager) =
  let itemsInQueueGauge = GaugeConf.create "Items_In_Queue" "help info with metric name" |> logManager.metricRegistry.registerMetric

  member x.enqueue item =
    itemsInQueueGauge.noLabels.inc 1.
    // enqueu logic

type C() =
  let itemsInQueueGauge = GaugeConf.create "Items_In_Queue" "help info with metric name" |> LogManager.DefaultMetricRegistry.registerMetric

  member x.enqueue item =
    itemsInQueueGauge.noLabels.inc 1.
    // enqueu logic

```

if user use A style, then concurrent is no need, but B/C style usage need ensure same metric name get same metric.